plugin/secrets_service_account_key.go

```diff
@@ -174,26 +174,17 @@ func (b *backend) secretKeyRevoke(ctx context.Context, req *logical.Request, d *
 }
 
 func (b *backend) getSecretKey(ctx context.Context, s logical.Storage, rs *RoleSet, keyType, keyAlgorithm string, ttl int) (*logical.Response, error) {
-	cfg, err := getConfig(ctx, s)
-	if err != nil {
-		return nil, errwrap.Wrapf("could not read backend config: {{err}}", err)
-	}
-	if cfg == nil {
-		cfg = &config{}
+	if rs.AccountId == nil {
+		return nil, fmt.Errorf("unable to create secret (service account key), roleset has nil account ID")
 	}
 
 	iamC, err := b.IAMAdminClient(s)
 	if err != nil {
 		return nil, errwrap.Wrapf("could not create IAM Admin client: {{err}}", err)
 	}
 
-	account, err := rs.getServiceAccount(iamC)
```

Comment:
Removing this - we did this for access tokens because it was potentially killing user's IAM quota. While I don't think the same issue happens with service account keys (since hard limit of 10 per service account), there's no real reason to have this check. keys.Create will 404 if the service account doesn't exist so we don't gain much error message